synapse/util/caches/descriptors.py

```diff
@@ -271,7 +271,7 @@ def __init__(
         else:
             self.function_to_call = orig
 
-        arg_spec = inspect.getargspec(orig)
+        arg_spec = inspect.getfullargspec(orig)
```

Comment:
Why ooi?

Reply:
Right, I was going to add a comment explaining it.  It's because `_get_bare_e2e_cross_signing_keys_bulk` has type annotations, so `getargspec` doesn't work on it, and it suggests to use `getfullargspec` instead.

Reply:
aaaaaah, cool